plugins/notifications/src/components/UserNotificationSettingsCard/UserNotificationSettingsPanel.tsx

```diff
@@ -109,6 +112,18 @@ export const UserNotificationSettingsPanel = (props: {
     );
   }
 
+  const uniqueOriginsMap = settings.channels
+    .flatMap(channel => channel.origins)
+    .reduce((map, origin) => {
+      if (!map.has(origin.id)) {
+        map.set(origin.id, origin);
+      }
+      return map;
+    }, new Map<string, OriginSetting>())
```

Comment:
Looks like something that could be `new Map((...).map(origin => [origin.id, origin]))`, as in passing an iterable of entries to the map constructor instead of manual reduce. It might not actually be more readable though